### ISSUE TITLE
rtsp not working (e.g. plex home theater)

### ISSUE THREAD

**Issue by wolfgang-braun** [Cigaras/IPTV.bundle#62]
Hi

I replaced one playlist entry using a rtsp url (cable tv stream from avm fritzbox router).

Plex home theater says "unrecognized file format"

In the IPTV log I found the following line:

`2016-11-19 15:22:42,089 (10950a000) :  DEBUG (services:618) - No matching services found for 'rtsp://192.168.178.254:554/?freq=322&bw=8&msys=dvbc&mtype=256qam&sr=6900&specinv=0&pids=0,16,17,18,20,100,1279,1104,1280,1285,4614'`

**Comment by f3doraco** [comment]
Can confirm, (almost) same device and get the same error.

The only difference is that i use the WLAN repeater instead of the router, the tuner and protocol is the same.

I think it used to work a while ago, after i tried again it didn't work anymore. I am not sure if it was the plex version, the IPTV plugin version or a combination of both (just using an older version of the plugin doesn't fix it).

**Comment by wolfgang-braun** [comment]
hey @f3doraco,

I think the bug is caused by the "HTTPLiveStreamURL" (plex) method.
Maybe plex provides a "RTSPVideoURL" method.

Does plex provide a documentation?

**Comment by Cigaras** [comment]
Nope, [no documentation](https://forums.plex.tv/discussion/172923/plugin-development-documentation), no `RTSPVideoURL` method.

**Comment by f3doraco** [comment]
The rtsp stream works with VLC and ffmpeg by the way.

**Comment by Cigaras** [comment]
As of version [2.0.x](https://github.com/Cigaras/IPTV.bundle/releases) it is possible to define specific protocol with some [advanced attributes](https://github.com/Cigaras/IPTV.bundle/wiki/Playlist-structure-and-examples#advance-attributes), however according to [Plex Framework](https://forums.plex.tv/discussion/comment/1379420/#Comment_1379420) [RTSP](https://en.wikipedia.org/wiki/Real_Time_Streaming_Protocol) is not a valid option, possible options are [DASH](https://en.wikipedia.org/wiki/Dynamic_Adaptive_Streaming_over_HTTP), HTTP, [HLS](https://en.wikipedia.org/wiki/HTTP_Live_Streaming) and [RTMP](https://en.wikipedia.org/wiki/Real-Time_Messaging_Protocol).
